src/System.Windows.Forms/src/System/Windows/Forms/Application.cs

```diff
@@ -1204,23 +1207,48 @@ public static void SetDefaultFont(Font font)
         if (NativeWindow.AnyHandleCreated)
             throw new InvalidOperationException(string.Format(SR.Win32WindowAlreadyCreated, nameof(SetDefaultFont)));
 
-        // If user made a prior call to this API with a different custom fonts, we want to clean it up.
-        if (s_defaultFont is not null && !ReferenceEquals(s_defaultFont, font))
-        {
-            s_defaultFont.Dispose();
-        }
-
         s_defaultFont = font;
         ScaleDefaultFont();
     }
 
+    /// <summary>
+    /// Scale <see cref="s_defaultFont"/> or <see cref="s_defaultFontScaled"/> if needed.
+    /// </summary>
     internal static void ScaleDefaultFont()
     {
-        // It is possible the existing scaled font will be identical after scaling the default font again. Figuring
-        // that out requires additional complexity that doesn't appear to be strictly necessary.
-        s_defaultFontScaled?.Dispose();
-        s_defaultFontScaled = null;
-        s_defaultFontScaled = ScaleHelper.ScaleToSystemTextSize(s_defaultFont);
+        if (s_defaultFont is null)
+            return;
```

Comment:
We typically add brackets
```suggestion
        if (s_defaultFont is null)
        {
            return;
        }
```